File: 8folds/box-plot.py
import pandas as pd
from plotnine import*


if __name__ == '__main__':
    independent_evaluation = pd.DataFrame()
    for i in range(8):
        independent_evaluation = pd.concat([independent_evaluation, pd.read_csv(f'group{i+1}/independent_evaluation/overall.csv')])
    independent_evaluation = independent_evaluation.reset_index().drop('index', axis=1)
    independent_evaluation.columns = ['Env', 'ROC-AUC', 'F-max']
    independent_evaluation['mode'] = 'independent'
    
    transfer_evaluation = pd.DataFrame()
    for i in range(8):
        transfer_evaluation = pd.concat([transfer_evaluation, pd.read_csv(f'group{i+1}/transfer_evaluation/overall.csv')])
    transfer_evaluation = transfer_evaluation.reset_index().drop('index', axis=1)
    transfer_evaluation.columns = ['Env', 'ROC-AUC', 'F-max']
    transfer_evaluation['mode'] = 'transfer'
    
    whole_evaluation = pd. concat([transfer_evaluation, independent_evaluation])
    # whole_evaluation.to_csv('plot-data.csv', index = 0)
    whole_evaluation = whole_evaluation.groupby('Env').filter(lambda x: ':V' in x)
    print(whole_evaluation)
    
    box_fig = (ggplot(whole_evaluation, aes(x='Env', y='ROC-AUC', fill='mode'))
               +geom_boxplot(show_legend=1)
               +xlim(['root:B','root:B:C','root:B:V','root:4M','root:4M:C','root:4M:V',
                           'root:12M','root:12M:C','root:12M:V','root:3Y','root:3Y:C','root:3Y:V',
                           'root:5Y','root:5Y:C','root:5Y:V','root:M']))
            
    print(box_fig)
        

File: 8folds/box-plot.R
library(ggplot2)
library(ggpubr)
plot.data = read.csv('plot-data.csv')

p1 = ggplot(plot.data, aes(x=Env, y=ROC.AUC, fill=mode))+
  geom_boxplot()+
  scale_fill_manual(values = c("#DC143C", "#87CEEB"))+
  xlim(c('NB(C)', 'NB(V)', '4M(C)', '4M(V)', 
         '12M(C)', '12M(V)', '3Y(C)', '3Y(V)', '5Y(C)', '5Y(V)','M'))+
  theme_bw()+
  xlab('')+
  ylab('AUROC')+
  stat_compare_means(aes(label = ..p.signif..), 
                     hide.ns = TRUE,
                     method = "wilcox.test",
                     label.y = 0.95,
                     size =10
                     )+
  theme(text=element_text(size = 10),
        axis.text = element_text(size = 10, color = 'black'),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        legend.background = element_blank(),
        legend.title = element_blank(),
        legend.position = c(0.92, 0.11))
p1

ggsave('r-plot-data.jpg', height = 100, width = 200, units = 'mm')
